src/components/SingleEvent/EventParticipants.js

```diff
@@ -49,26 +56,52 @@ class EventParticipants extends Component {
       spots = `${participants.length} out of ${participantLimit} attended`
     } else {
       const spotsLeft = participantLimit - participants.length
-      spots = `${participants.length} going, ${spotsLeft} ${pluralize('spot', spotsLeft)} left`
+      spots = `${participants.length} going, ${spotsLeft} ${pluralize(
+        'spot',
+        spotsLeft
+      )} left`
     }
 
     return (
       <Fragment>
-        <H3>Participants - <Spots>{spots}</Spots></H3>
+        <H3>
+          Participants - <Spots>{spots}</Spots>
+        </H3>
         <EventFilters
           handleSearch={this.handleSearch}
-          search={this.state.search }
+          handleFilterChange={this.handleFilterChange}
+          amAdmin={amAdmin}
+          search={this.state.search}
           enableQrCodeScanner={amAdmin}
+          ended={ended}
         />
         <EventParticipantsContainer>
-          { participants.length > 0 ? (
+          {participants.length > 0 ? (
             participants
               .sort((a, b) => (a.index < b.index ? -1 : 1))
-              .filter(p => (
-                (p.user.realName || '').toLowerCase().includes(lowerSearch) ||
-                (p.user.username || '').toLowerCase().includes(lowerSearch) ||
-                p.user.address.toLowerCase().includes(lowerSearch)
-              ))
+              .filter(p => {
```

Comment:
Am thinking the sort and filter logic should be shifted out into a first class function on this component, since at the moment we're basically reconstructing these functions on every render.

Reply:
Let's sort this out in a different PR? I want to make the filtering more modular anyway